crates/rpc-types/src/eth/block.rs

```diff
@@ -75,13 +75,17 @@ pub struct Header {
     /// Difficulty
     pub difficulty: U256,
     /// Block number
-    pub number: Option<U256>,
+    #[serde(default, with = "alloy_serde::num::u64_hex_opt")]
+    pub number: Option<u64>,
```

Comment:
`skip_serializing_if` and feels like this is mandatory too?

Reply:
number is acutally required, yes

Reply:
there are no clear rules for the pending block,
until we figure it out, let's keep them optional